From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bjarne Koll <lynxplay101@gmail.com>
Date: Thu, 16 Mar 2023 14:33:33 +0100
Subject: [PATCH] Improve schedule task information

When shutting down the server, tasks may not cancel properly due to some
plugin not closing their tasks.
Finding out which task is scheduled where and not closed however has
proven troublesome.

To fix this, this patch adds the ability to both plugins and servers to
fetch the implementation class type of overdue tasks from BukkitWorker
instances which are displayed in the nag message on shutdown.

diff --git a/src/main/java/org/bukkit/scheduler/BukkitWorker.java b/src/main/java/org/bukkit/scheduler/BukkitWorker.java
index 81bdc9ceafb2a7cd3b93f4049504b3d5de049536..227c695c6c89e8a3e6fa8f6b3f56da0e42390930 100644
--- a/src/main/java/org/bukkit/scheduler/BukkitWorker.java
+++ b/src/main/java/org/bukkit/scheduler/BukkitWorker.java
@@ -34,4 +34,12 @@ public interface BukkitWorker {
     @NotNull
     public Thread getThread();
 
+    // KTP start - improved task information
+    /**
+     * Provides the java class that was passed by a plugin to the server that is executed by this worker.
+     *
+     * @return the java class instance.
+     */
+    @NotNull Class<?> getTaskClass();
+    // KTP end - improved task information
 }
